src/runtime/pkg/katautils/config.go

```diff
@@ -131,7 +131,7 @@ type hypervisor struct {
 	MemSlots                       uint32          `toml:"memory_slots"`
 	DefaultBridges                 uint32          `toml:"default_bridges"`
 	Msize9p                        uint32          `toml:"msize_9p"`
-	NumVCPUs                       int32           `toml:"default_vcpus"`
+	NumVCPUs                       float64         `toml:"default_vcpus"`
```

Comment:
Can we use float32 here just to avoid using more memory? I can't see us exceeding 2^23 CPUs anytime soon.

Reply:
@c3d, thanks for the suggestion, and I do believe it makes sense.

I did opt for using `float64` instead of `float32` because the golang's `Ceil` function takes a `float64` as parament and returns a `float64`.  Meaning that we'd need to cast it to `float32` in a lot of places.

Do you think it's still worh doing the casts here and there, to avoid using `float64`?

Reply:
> For example, consider [the int cast in (MacvtapEndpoint) Attach](https://github.com/kata-containers/kata-containers/blob/708b0a305267d5ae0641122e55576067d107182a/src/runtime/virtcontainers/macvtap_endpoint.go#L80)

Very nice catch!

Reply:
@fidencio what do you think of [this approach](https://github.com/c3d/kata-containers/tree/topic/runtime-improve-vcpu-allocation-on-host-side). It looks like in practice, the casts are more from `uint32` than from `float64`, and most of the `float` casts replace `float64` casts with `float32`. So I believe in the end it is worth it. WDYT?

Reply:
LGTM @c3d. I've just commented about some oversights in your proposal at https://github.com/kata-containers/kata-containers/commit/285f4de51ff937141b077a1475964e787fe9dca1.

I'd recommend to squash the patches for the sake of bisect.